pkg/scheduler/scheduler.go

```diff
@@ -68,7 +68,7 @@ func (self *slaveStorage) getSlaveIds() []string {
 	self.Lock()
 	defer self.Unlock()
 	slaveIds := make([]string, 0, len(self.slaves))
-	for slaveID, _ := range self.slaves {
+	for slaveID := range self.slaves {
```

Comment:
Was this code correct? It seems we were collecting the indexes, not the values in the slice.